### PR TITLE
Fix false positive for appointment clash check

### LINKED ISSUE #376: [PE-D][Tester D] Unable to add consecutive appointments
In the UG it states that appointment times are exclusive but I am unable to add consecutive appointments for the same patient.

-------------
Labels: `severity.Medium` `type.FunctionalityBug`
original: yuvrajaryan/ped#8

### PR DESCRIPTION
**Pull request by yadobler**
fixes #376

### CODE REVIEW

File: src/main/java/seedu/address/model/appointment/Appointment.java [hunk 1]
```diff
@@ -111,24 +111,42 @@ private LocalTime getTimeFromString(String time) {
...
+        // -----|--^---|-------
+        //     st     et
+        boolean isApptStartingDuringTimePeriod = appointmentStartTime.isAfter(startTime)
+                                               && appointmentStartTime.isBefore(endTime);
+
+        // ast------------aet
+        // -----|--^--|-------
+        //     st     et
+
+        boolean isApptEnclosing = !appointmentEndTime.isBefore(endTime)
```
Comment (yong-zhuo) [inline src/main/java/seedu/address/model/appointment/Appointment.java:146]: Good checks for the different possible scenarios 👍🏻

File: src/test/java/seedu/address/model/appointment/AppointmentTest.java [hunk 2]
```diff
@@ -143,27 +143,45 @@ public void testEquals() throws IllegalValueException {
...
+        Assertions.assertFalse(a.isClashing("2024-10-29", "0900-1235")); // "laterDateEndInside"
+        Assertions.assertFalse(a.isClashing("2024-10-29", "1000-1235")); // "laterDateSameTime"
+        Assertions.assertFalse(a.isClashing("2024-10-29", "1000-1200")); // "laterDateInside"
+        Assertions.assertFalse(a.isClashing("2024-10-29", "1100-1200")); // "laterDateInside"
+        Assertions.assertFalse(a.isClashing("2024-10-29", "1100-1235")); // "laterDateInside"
+        Assertions.assertFalse(a.isClashing("2024-10-29", "1000-1500")); // "laterDateEnclose"
+        Assertions.assertFalse(a.isClashing("2024-10-29", "1000-1500")); // "laterDateStartInside"
+        Assertions.assertFalse(a.isClashing("2024-10-29", "1234-1500")); // "laterDateStartInside"
+        Assertions.assertFalse(a.isClashing("2024-10-29", "1235-1300")); // "laterDateAfterConseq"
+        Assertions.assertFalse(a.isClashing("2024-10-29", "1300-1500")); // "laterDateAfter"
```
Comment (yong-zhuo) [inline src/test/java/seedu/address/model/appointment/AppointmentTest.java:185]: Very comprehensive testing for each scenario the user can encounter

**Review by yong-zhuo — APPROVED** [review]
Overall LGTM to merge

**Review by PresleyChew — APPROVED** [review]
Overall LGTM

**Pull request merged**